### PR TITLE
fix dockerfile local dir

### PR DESCRIPTION
**Pull request by junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

Fixes: https://github.com/containerd/nerdctl/issues/1191

In addition, check in advance whether the dockerfile file exists.

before:
```
[N] ❯ sudo nerdctl build -t backstage:dev ..
[+] Building 0.0s (2/2) FINISHED
 => [internal] load .dockerignore                                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                                             0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                        0.0s
 => => transferring dockerfile: 2B                                                                                                                                                                                          0.0s
error: failed to solve: failed to read dockerfile: open /tmp/buildkit-mount1687483572/Dockerfile: no such file or directory
FATA[0000] unrecognized image format
```

after:
```
[N] ❯ sudo nerdctl build -t backstage:dev ..
FATA[0000] lstat /home/jun/Documents/workspace/my-backstage/packages/Dockerfile: no such file or directory
```

### CODE REVIEW

**Review by manugupt1 — APPROVED** [review]
thanks for the changes!

File: cmd/nerdctl/build_test.go [hunk 1]
```diff
@@ -140,6 +140,38 @@ CMD ["echo", "nerdctl-build-test-stdin"]
 	base.Cmd("build", "-t", imageName, "-f", "-", ".").CmdOption(testutil.WithStdin(strings.NewReader(dockerfile))).AssertCombinedOutContains(imageName)
 }
 
+func TestBuildWithDockerfile(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").Run()
```
[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/build_test.go:146]: cc @ktock

**Review by fahedouch — APPROVED** [review]

File: cmd/nerdctl/build_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktock) [reply cmd/nerdctl/build_test.go:146]: @manugupt1 @junnplus This was to ensure all content (possibly leased by BuildKit) in the content store gets pruned after each test. Tests may check the behaviour of pulling blobs and contents remaining in the content store unexpectedly skip these operations.But yes, we may improve this by removing `defer base.Cmd("builder", "prune").Run()` and instead adding `rmiAll()` (defined in `image_encrypt_linux_test.go`) to tests (if exist) requiring clean content store. Thanks for pointing this out :)

**Review by Zheaoli — APPROVED** [review]
LGTM

File: cmd/nerdctl/build_test.go [hunk 2]
```diff
@@ -140,6 +140,38 @@ CMD ["echo", "nerdctl-build-test-stdin"]
...
+	err = os.WriteFile(filepath.Join(buildCtx, "Dockerfile"), []byte(dockerfile), 0644)
+	assert.NilError(t, err)
+
+	pwd, err := os.Getwd()
+	assert.NilError(t, err)
+	err = os.Chdir(buildCtx)
+	assert.NilError(t, err)
+	defer os.Chdir(pwd)
+
+	// hack os.Getwd return "(unreachable)" on rootless
```
Comment (AkihiroSuda) [inline cmd/nerdctl/build_test.go:166]: How does this happen?

Can't find "unreachable" error in the code
https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/getwd.go;l=22

Comment (junnplus) [reply cmd/nerdctl/build_test.go:166]: I actually don't know what happened, `Getwd` returns a strange path during rootless testing.

https://github.com/containerd/nerdctl/runs/7196804919?check_suite_focus=true#step:5:1004

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**